src/gluonts/model/deepar/_network.py

```diff
@@ -335,6 +359,10 @@ def distribution(
             future_target=future_target,
         )
 
+        # store output of rnn layers, so that it can be used for regularization later
+        # assume no dropout for outputs, so can be directly used for activation regularization
+        self.rnn_outputs = rnn_outputs
```

Comment:
It is better to return `rnn_outputs` along with `distribution`. If it is only needed for training you could take in additional argument `return_rnn_outputs=True/False` and return either `distribution` or both depending on that input.